### DISCUSSION TITLE
Running an application outside of core run

### DISCUSSION THREAD

**Discussion by runrc** [cogentcore/core#994] [Q&A] [answered]
We would like to execute a cogent core GUI application from outside of `core run`

Use cases are,
1) Run via `go run`
2) Run from an IDE for debugging purposes
3) Run `-race` detection

How can we do the above?

**Comment by kkoreilly** [comment, marked as answer]
As stated [here](https://www.cogentcore.org/core/basics/building) on the Cogent Core website, you can always build and run Cogent Core apps using the standard Go commands (`go build`, `go run`, etc). This includes running from an IDE using a debugger and running the race detector (there is more information on running the race detector in #671).

You need use `core run` and other `core` commands such as `core build` and `core pack` when building for mobile and web and creating published release builds. You should also use it in general when possible since it reduces binary sizes and bundles app resources and information. However, it is unnecessary when building during development for things like debugging and race detection.
